### DISCUSSION TITLE
How to specify colorbar keyword arguments when using quickplot.pcolormesh?

### DISCUSSION THREAD

**Discussion by krikru** [SciTools/iris#5964] [Q&A] [answered]
When using [iris.quickplot.pcolormesh](https://scitools-iris.readthedocs.io/en/v3.0.0/generated/api/iris/quickplot.html#iris.quickplot.pcolormesh), a colorbar is created automatically. How can I specify keywords arguments for that colorbar, like I can do when I create a colorbar manually and add it to a plot created with [iris.plot.pcolormesh](https://scitools-iris.readthedocs.io/en/v3.0.0/generated/api/iris/plot.html#iris.plot.pcolormesh)?

**Comment by krikru** [comment]
Or is there some better way to solve it in than the way I’m trying to solve it in while still getting the same behavior?

**Comment by rcomer** [comment]
Which keywords do you want to pass?  Some properties of the [`Colorbar`](https://matplotlib.org/stable/api/colorbar_api.html#matplotlib.colorbar.Colorbar) can be set afterwards via its `set_...` methods, e.g.

```python
import iris.quickplot as qplt

mesh = qplt.pcolormesh(my_data)
mesh.colorbar.set_label('stuff')
```

For full control, I think the right way is to use `iris.plot.pcolormesh` and add the colorbar manually.  I think quickplot was only really designed for a "first look" at your data and not with plot configuration in mind.

**Reply by krikru** [reply]
Thanks for your reply!

The function I want to pass the keyword arguments in just takes an argument `colorbar_kwargs` as an argument which I had hoped that I could pass on. But using `iris.plot.pcolormesh` instead and creating a colorbar manually may indeed be the most proper way to go about it.

So in order to get the same behavior with `iris.plot.pcolormesh` as I would get with `iris.quickplot.pcolormesh` (which I want to for consistency purposes), I have noticed that there are a couple of things that I need to do. First I need to create and add a colorbar, which I know how to do. Then I also need to add a label, since `iris.quickplot.pcolormesh` seems to add the units of the cube as a label to the colorbar. Where in the code does that happen, so I can look at how a label is created from the units? Looking at the pieces of code for `iris.plot.pcolormesh` and `iris.quickplot.pcolormesh` respectively, though, there seems like they are quite different. Are there more differences between them that I should know of, except for that that the quicklplot version adds a colorbar and a label for the colorbar?

**Reply by trexfeathers** [reply]
Is this what you are looking for?

https://github.com/SciTools/iris/blob/3e1e80ecdb3784f25f4f2bc03bc8e0dbb0a1afa0/lib/iris/quickplot.py#L100-L116

**Reply by rcomer** [reply, marked as answer]
The above snippet gives the plot title and the labels for the x- and y-axis, which quickplot also adds automatically (although the x- and y- labels are skipped for maps).  The colorbar label is defined here:
https://github.com/SciTools/iris/blob/3e1e80ecdb3784f25f4f2bc03bc8e0dbb0a1afa0/lib/iris/quickplot.py#L66-L69

**Comment by trexfeathers** [comment]
https://github.com/SciTools/iris/issues/5970

**Comment by krikru** [comment]
> The above snippet gives the plot title and the labels for the x- and y-axis, which quickplot also adds automatically (although the x- and y- labels are skipped for maps). The colorbar label is defined here:
> 
> https://github.com/SciTools/iris/blob/3e1e80ecdb3784f25f4f2bc03bc8e0dbb0a1afa0/lib/iris/quickplot.py#L66-L69

Thanks; this is probably what I was looking for since I'm plotting maps, but the code for adding x- and y-labels could definitely be useful too.

How does Iris know when a map is being plotted? Does is just check whether the two axis coordinates are spatial coordinates?

**Comment by krikru** [comment]
Thank you both anyway for your answers. I think I have my code the way  want it now.
